microservices/biobuddy/biobuddy/gpt_researcher/retrievers/tavily_search/tavily_search.py

```diff
@@ -0,0 +1,51 @@
+# Tavily API Retriever
+
+# libraries
+import os
+from tavily import TavilyClient
+from duckduckgo_search import DDGS
+
+
+class TavilySearch():
+    """
+    Tavily API Retriever
+    """
+    def __init__(self, query):
+        """
+        Initializes the TavilySearch object
+        Args:
+            query:
+        """
+        self.query = query
+        self.api_key = self.get_api_key()
+        self.client = TavilyClient(self.api_key)
+
+    def get_api_key(self):
+        """
+        Gets the Tavily API key
+        Returns:
+
+        """
+        # Get the API key
+        try:
+            api_key = os.environ["TAVILY_API_KEY"]
+        except:
+            raise Exception("Tavily API key not found. Please set the TAVILY_API_KEY environment variable. "
```

Comment:
Shall we make tavily optional (i.e. in case TAVILY_API_KEY is not set then ignore this retriever)? In case that a user does not want to use tavily but still wishes to run biobuddy reports